invenio_oauthclient/contrib/cern.py

```diff
@@ -209,16 +211,68 @@ def account_setup(remote, token):
 
     response = remote.get(REMOTE_APP_RESOURCE_API_URL)
     user = token.remote_account.user
+    if not user:
+        raise Exception('`user` should not be None, something bad happened!')
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
+
+        # update session object
+        groups_old = current_user.info['group']
+        groups_new = fetch_groups(res['Group'])
+        current_user.info['group'] = groups_new
```

Comment:
The original usage of having a copy of CERN egroups into Invenio was that then it was possible to do e.g. messaging/notifications to groups, independently to the single user being authenticated.